### PR TITLE
🔧 Unit test onCreateProposal Cloud Function

### PR DESCRIPTION
**Pull request by corinnekrych**

### CODE REVIEW

**Comment by corinnekrych** [comment]
This PR add:
- `npm run test:coverage` to run test with coverage
- test for all triggered functions

File: functions/triggers/onCreateProposal.test.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  const snap = {
+    data: () => talk,
+  }
+  // mock email.send
+  let emailSend
+  conf = { mailgun: { domain: 'somedomain.org', key: 'SOME-SECRET' }, app: { url: 'https://somefirebase.url' } }
+  beforeEach(() => {
+    test.mockConfig(conf)
+    // mock firestore DB static methods
+    Object.defineProperty(admin, 'firestore', { get: () => databaseStub, configurable: true })
```
Comment (frinyvonnick) [inline functions/triggers/onCreateProposal.test.js:59]: Maybe you could put this code in a function `mockFirestoreMethods` at the end of the file (thanks hoisting in JS ❤️) so the first thing we see when opening the file are actual tests and not configuration ? What do you think about it ? 😄

File: functions/triggers/onUpdateProposal.js [hunk 2]
```diff
@@ -4,11 +4,17 @@ const { getEvent } = require('../firestore/event')
 const { getUsers } = require('../firestore/user')
 const { updateProposal } = require('../firestore/proposal')
 const { partialUpdateTalk } = require('../firestore/talk')
-const sendEmail = require('../email')
+const email = require('../email')
 const talkAccepted = require('../email/templates/talkAccepted')
 const talkRejected = require('../email/templates/talkRejected')
 
-
+// onUpdateProposal is called when a submission is updated.
```
Comment (frinyvonnick) [inline functions/triggers/onUpdateProposal.js:11]: Maybe we could use jsdoc for this kind of comments ? Then we would be able to generate a doc for contributors with https://github.com/jsdoc3/jsdoc. What do you think about it ?

**Review by bpetetot — APPROVED** [review]
Thanks @corinnekrych. Awesome work 👍

File: functions/triggers/onCreateProposal.test.js (hunk 1 above)
Comment (corinnekrych) [reply functions/triggers/onCreateProposal.test.js:59]: Hoisting as a beloved feature :)
why not? the mocking is quite heavy I'd agree

File: functions/triggers/onUpdateProposal.js (hunk 2 above)
Comment (corinnekrych) [reply functions/triggers/onUpdateProposal.js:11]: could we use docz for functions too? @bpetetot

**Comment by corinnekrych** [comment]
@bpetetot feel free to merge, we can always improve doc later :)

**Comment by bpetetot** [comment]
Sorry for the delay

**Pull request merged**